src/app/rpmostree-ex-builtin-commit2jigdo.c

```diff
@@ -0,0 +1,1167 @@
+      const char *checksum;
+      ostree_object_name_deserialize (object, &checksum, &objtype);
+      if (objtype == OSTREE_OBJECT_TYPE_FILE)
+        g_hash_table_add (self->commit_content_objects, g_strdup (checksum));
+      g_hash_table_iter_remove (&it);
+    }
+  g_print ("%u content objects\n", g_hash_table_size (self->commit_content_objects));
+
+  g_print ("Finding reachable objects from packages...\n");
+  g_autoptr(RpmOstreeRefSack) sack =
+    rpmostree_get_refsack_for_commit (self->repo, commit, cancellable, error);
+  if (!sack)
+    return FALSE;
+
+  hy_autoquery HyQuery hquery = hy_query_create (sack->sack);
+  hy_query_filter (hquery, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
+  g_autoptr(GPtrArray) pkglist = hy_query_run (hquery);
+  g_print ("Building object map from %u packages\n", pkglist->len);
+
+  g_assert_cmpint (pkglist->len, >, 0);
+
+  for (guint i = 0; i < pkglist->len; i++)
+    {
+      DnfPackage *pkg = pkglist->pdata[i];
+      if (!build_objid_map_for_package (self, pkg, cancellable, error))
+        return FALSE;
+    }
+
+  g_print ("%u content objects in packages\n", g_hash_table_size (self->content_object_to_pkg_objid));
+  g_print ("  %u duplicate, %u unused\n",
+           self->n_duplicate_pkg_content_objs, self->n_unused_pkg_content_objs);
+  g_print ("  %u big sizematches, %u/%u nonunique basenames\n",
+           self->duplicate_big_pkgobjects, self->n_nonunique_objid_basenames, self->n_objid_basenames);
+  g_autoptr(GHashTable) new_reachable_big = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+  g_autoptr(GHashTable) new_reachable_small = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+  g_autoptr(GHashTable) pkgs_with_content = g_hash_table_new (NULL, NULL);
+  guint64 pkg_bytes = 0;
+  guint64 oirpm_bytes_small = 0;
+
+  /* Loop over every content object in the final commit, and see whether we
+   * found a package that contains that exact object.  We classify new
+   * objects as either "big" or "small" - for "big" objects we'll try
+   * to heuristically find a content-identical one.
+   */
+  GLNX_HASH_TABLE_FOREACH (self->commit_content_objects, const char *, checksum)
+    {
+      guint32 objsize;
+      if (!query_objsize_assert_32bit (self->repo, checksum, &objsize, error))
+        return FALSE;
+      const gboolean is_big = objsize >= BIG_OBJ_SIZE;
+
+      PkgObjid *pkgobjid = g_hash_table_lookup (self->content_object_to_pkg_objid, checksum);
+      if (!pkgobjid)
+        g_hash_table_add (is_big ? new_reachable_big : new_reachable_small, g_strdup (checksum));
+      else
+        g_hash_table_add (pkgs_with_content, pkgobjid->pkg);
+
+      if (pkgobjid)
+        pkg_bytes += objsize;
+      else if (!is_big)
+        oirpm_bytes_small += objsize;
+      /* We'll account for new big objects later after more analysis */
+    }
+
+  g_print ("%u/%u packages contain objects; %u small, examining %u big objects more closely...\n",
```

Comment:
This is minor, though I'd word this like:

```
Found objects in 123/456 packages; found $small + $big objects unpackaged.
```

Also, it seems like the big objects are examined lower down after this next block, right?